### ISSUE TITLE
[🐛 Bug]: New project does run with `wdio` cli but does not compile with `tsc`

### ISSUE THREAD

**Issue by kmile** [webdriverio/webdriverio#9873]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.3.10

### Node.js Version

16.18.1

### Mode

WDIO Testrunner

### What happened?

I started a new project by adding `@wdio/cli` to my project. Then I ran `wdio` to set up a new dummy test case, using Typescript, on Node 16.

When that was completed I am able to run the test case successfully. However, when I try to compile the project using `tsc` I get a number of errors:

```
node_modules/webdriverio/build/commands/element/nextElement.d.ts:24:72 - error TS2307: Cannot find module '../..' or its corresponding type declarations.

24. export declare function nextElement(this: WebdriverIO.Element): import("../..").ChainablePromiseElement<WebdriverIO.Element>;
```

And the same for `parentElement` and `previousElement`.  I had to change `moduleResolution` and `allowSyntheticDefaultImports` in my default `tsconfig.json` to resolve a couple other errors, but this one I cannot resolve.

If this is not a problem in my setup, can the `cli` generate a config that will actually build from the start, without manual modifications and troubleshooting?

### What is your expected behavior?

A default project created with `@wdio/cli` can run `tsc` without having to make any adjustments.

### How to reproduce the bug.

```
npm init
npm install @wdio/cli
```
Change `scripts/test` to `wdio` in `package.json`

```
npm run test
```

* Create config file
* Pick Local
* Pick Local machine
* Pick Mocha
* Pick Typescript
* Generate Test Files
* Default reporter
* No plugins
* Pick chromedriver
* Run `npm install`

`tsc` will fail on module resolution right now.

* Change `tsconfig.json`:
* Set `moduleResolution` to `node16`
* Add `allowSyntheticDefaultImports` to `true`
* Add `"type": "module"` to `package.json`

Try `tsc` again.

See above errors in output.

### Relevant log output

```typescript
node_modules/webdriverio/build/commands/element/nextElement.d.ts:24:72 - error TS2307: Cannot find module '../..' or its corresponding type declarations.

24. export declare function nextElement(this: WebdriverIO.Element): import("../..").ChainablePromiseElement<WebdriverIO.Element>;
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Are there any specific reasons why you have your `tsconfig` like this? Also is it important for the tests hat TypeScript can compile them with given deps? I am getting errors from other packages, e.g. Puppeteer as well.

One solution to this could be to explicitly set a return value here (`ChainablePromiseElement`). This needs to be done for all commands that return elements.

**Comment by kmile** [comment]
Thanks for looking into this!

This is the `tsconfig` that was generated by the `wdio` initialization script (I assume) since I didn't create it and it showed up after initialization. I've done nothing in the project except run `wdio` and make the changes in the reproduction steps to get rid of some errors that had obvious recommendations.

The reason I am running `tsc` is that our test package is a separate package, and our tool chain wants to compile it by default. Given the errors, I assumed I messed something up in my configuration, some dependencies got mismatched, or there was some problem I'd run into down the line once I want to start using those operators whose types can't be resolved.

I'd prefer not to make changes to the `node_modules` since then I'd have to maintain it. I could try and add explicit types to my project itself, but 1) I wonder if I can even help it resolve `../../`, and 2) I'd rather not maintain a separate type, but instead help resolve it explicitly in `wdio` so others don't run into the same thing.

If the `wdio` types are not meant to be compiled using `tsc`, and that's not a problem using `wdio` or any of the missing/broken types down the line, then I guess I'll try and avoid compiling the test package :)

**Comment by Jmcosel** [comment]
@kmile I am solving this problem (at least with the auto-generation of files) with PR #9771 , in case you want to look at that and tell me if you think it addresses what you experienced.

**Comment by kmile** [comment]
The changes look good, but I don't know enough  about the codebase to know if this will solve my exact issue. I'm happy to test again when a new version is available, or if I can point my local install to a branch or something.

**Comment by christian-bromann** [comment]
@kmile changes were shipped with [`v8.5.7`](https://github.com/webdriverio/webdriverio/commit/b8b610be11417ada5809f75266f0462abeb5df79) can you test with that version again?

**Comment by christian-bromann** [comment]
Closing due to inactivity. Happy to re-open if issue still appears. Thank you for the fix @Jmcosel
